google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/AuthorizedView.java

```diff
@@ -0,0 +1,309 @@
+     * to be used by applications.
+     */
+    @InternalApi
+    public static SubsetView fromProto(
+        @Nonnull com.google.bigtable.admin.v2.AuthorizedView.SubsetView proto) {
+      return new SubsetView(proto);
+    }
+
+    private SubsetView(@Nonnull com.google.bigtable.admin.v2.AuthorizedView.SubsetView proto) {
+      this.builder = proto.toBuilder();
+    }
+
+    public SubsetView() {
+      this.builder = com.google.bigtable.admin.v2.AuthorizedView.SubsetView.newBuilder();
+    }
+
+    /** Gets the row prefixes to be included in this subset view. */
+    public ImmutableList<ByteString> getRowPrefixes() {
+      return ImmutableList.copyOf(this.builder.getRowPrefixesList());
+    }
+
+    /** Gets the map from familyId to familySubsets in this subset view. */
+    public ImmutableMap<String, FamilySubsets> getFamilySubsets() {
+      ImmutableMap.Builder<String, FamilySubsets> familySubsets = ImmutableMap.builder();
+      for (Entry<String, com.google.bigtable.admin.v2.AuthorizedView.FamilySubsets> entry :
+          builder.getFamilySubsetsMap().entrySet()) {
+        familySubsets.put(entry.getKey(), FamilySubsets.fromProto(entry.getValue()));
+      }
+      return familySubsets.build();
+    }
+
+    /** Adds a new rowPrefix to the subset view. */
+    public SubsetView addRowPrefix(ByteString rowPrefix) {
+      this.builder.addRowPrefixes(rowPrefix);
+      return this;
+    }
+
+    /** Adds a new rowPrefix to the subset view. */
+    public SubsetView addRowPrefix(String rowPrefix) {
+      this.builder.addRowPrefixes(ByteString.copyFromUtf8(rowPrefix));
+      return this;
+    }
+
+    /**
+     * Adds a new familyId with its familySubsets to the subset view. Please note that calling this
+     * method with the same familyId will overwrite the previous rule set on the family.
+     */
+    public SubsetView addFamilySubsets(String familyId, FamilySubsets familySubsets) {
+      this.builder.putFamilySubsets(familyId, familySubsets.toProto());
+      return this;
+    }
+
+    /**
+     * Creates the request protobuf. This method is considered an internal implementation detail and
+     * not meant to be used by applications.
+     */
+    @InternalApi
+    public com.google.bigtable.admin.v2.AuthorizedView.SubsetView toProto() {
+      return builder.build();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      SubsetView that = (SubsetView) o;
+      return Objects.equal(builder.build(), that.builder.build());
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hashCode(builder.build());
+    }
+  }
+
+  /** Represents subsets of a particular column family that are included in this authorized view. */
+  public static final class FamilySubsets {
```

Comment:
can we separate each of the `public static final class`es into their own files? This is making this one file really large.

Reply:
Hmm these classes are only useful in the context of AuthorizedViews so I would prefer keeping those in the `AuthorizedView.java`.
Also given the urgency of this PR, can we keep it this way for now, since we can always do the refactor later?

Reply:
moving the classes would be considered a breaking change, so we should decide now. @mutianf do you have a preference?

Reply:
Maybe we can move SubsetView and FamilySubset out to a new class? WE might add more AuthorizedViewTypes in the future, so maybe it's better to move them out now and avoid this file get even larger ?

Reply:
Sure. Just to confirm, do you mean to move the SubsetView and FamilySubset to two separate files?